src/block_loss_functions.rs

```diff
@@ -32,95 +34,153 @@ pub fn logistic(t: f32) -> f32 {
 
 
 pub struct BlockSigmoid {
+    num_inputs: usize,
+    input_offset: usize,
+    output_offset: usize,
+    copy_to_result: bool
 }
 
+
+pub fn new_logloss_block(  bg: &mut graph::BlockGraph, 
+                           input: graph::BlockPtrOutput,
+                           copy_to_result: bool) 
+                        -> Result<graph::BlockPtrOutput, Box<dyn Error>> {    
+    let num_inputs = bg.get_num_output_values(vec![&input]);
+    let block = Box::new(BlockSigmoid {num_inputs: num_inputs as usize,
+                                input_offset: usize::MAX,
+                                output_offset: usize::MAX,
+                                copy_to_result: copy_to_result});
+    let mut block_outputs = bg.add_node(block, vec![input]).unwrap();
+    assert_eq!(block_outputs.len(), 1);
+    Ok(block_outputs.pop().unwrap())
+
+}
+
+
+pub fn new_without_weights(mi: &model_instance::ModelInstance, 
+                            num_inputs: u32,
+                            copy_to_result: bool) -> Result<Box<dyn BlockTrait>, Box<dyn Error>> {
+    Ok(Box::new(BlockSigmoid {num_inputs: num_inputs as usize,
+                                input_offset: usize::MAX,
+                                output_offset: usize::MAX,
+                                copy_to_result: copy_to_result}))
+}
+
+
 impl BlockTrait for BlockSigmoid {
 
     fn as_any(&mut self) -> &mut dyn Any {
         self
     }
 
-    fn new_without_weights(mi: &model_instance::ModelInstance) -> Result<Box<dyn BlockTrait>, Box<dyn Error>> {
-        Ok(Box::new(BlockSigmoid {}))
+    fn get_num_output_slots(&self) -> usize {1}   
+
+
+    fn get_num_output_values(&self, output: graph::OutputSlot) -> usize {
+        assert!(output.get_output_index() == 0);
+        1
+    }
+    
+    fn set_input_offset(&mut self, input: graph::InputSlot, offset: usize)  {
+        assert!(input.get_input_index() == 0);
```

Comment:
Assertions are in Rust checked also in release builds, we could try without at some point (if not critical function this does not matter of course)